### ISSUE TITLE
change drag & drop icon depending on action

### ISSUE THREAD

**Issue by avilleret** [ossia/score#46]
When I drag & drop value from namespace into my scenario I can either update a state or add a new one.
I should be very nice that the icon represent the upcomming task and change depending on what is below the mouse.

**Comment by bltzr** [comment]
yes, and more generally context-dependant mouse icons would be great. how hard is this @jcelerier ?

**Comment by jcelerier** [comment]
not very hard

**Comment by jcelerier** [comment]
the most important I think would be for someone to make a set of icons because currently they would look different on every system (and not all systems have the same set of icons)

**Comment by bltzr** [comment]
@jln- do you feel in the mood for some cursor icon design ? (not necessarily today or next week though)

**Comment by jcelerier** [comment]
## To do : 

* Look in the code for classes that have drag actions.
  * For instance : classes that inherit from QTreeView, QListView, such as DeviceExplorerView.
  * The documentation about this can be interesting to read : http://doc.qt.io/qt-5/model-view-programming.html#using-drag-and-drop-with-item-views
* Reimplement startDrag and push a new cursor with `QApplication::setOverrideCursor`
* Connect `QApplication::restoreOverrideCursor` to the `destroyed` signal of the QDrag.

**Comment by jln-** [comment]
Sorry, was away for the last 3 weeks.

Yep, I'll give it a try.

**Comment by jcelerier** [comment]
the cursors are done by @aklevy, just need to integrate them
